extensions/common/vault/vault-aws/src/main/java/org/eclipse/edc/vault/aws/AwsSecretsManagerVaultDefaultSanitationStrategy.java

```diff
@@ -25,34 +26,41 @@ public AwsSecretsManagerVaultDefaultSanitationStrategy(Monitor monitor) {
 
     /**
      * Many-to-one mapping from all strings into set of strings that only contains valid AWS Secrets Manager key names.
-     * The implementation replaces all illegal characters with '_' and attaches the hash code of the original string to
-     * minimize the likelihood of key collisions.
+     * The implementation replaces all illegal characters with '_' and attaches the hash code of the original string,
+     * when illegal characters are replaced, to minimize the likelihood of key collisions.
      *
      * @param originalKey any key
      * @return Valid AWS Secrets Manager key
      */
     @Override
     public String sanitizeKey(String originalKey) {
         var key = originalKey;
-        if (originalKey.length() > 500) {
+        boolean originalKeyReplaced = false;
+
+        if (originalKey.length() > 512) {
+            //Substring to 500, to add the 12 char hashcode
             key = originalKey.substring(0, 500);
+            originalKeyReplaced = true;
         }
+
         var sb = new StringBuilder();
-        boolean replacedIllegalCharacters = false;
         for (int i = 0; i < key.length(); i++) {
             var c = key.charAt(i);
             if (!Character.isLetterOrDigit(c) && c != '/' && c != '_' && c != '+' && c != '.' && c != '@' && c != '-') {
-                replacedIllegalCharacters = true;
+                originalKeyReplaced = true;
                 sb.append('-');
```

Comment:
the documentation says `_` but the code does `-`. Is this an error?

Reply:
This logic is for replacing invalidad characters in the original key to "-". (ex: example#key, replaced to example-key)
The hashcode is appended afterwards.